redis/cache/redispipebp/monitored.go

```diff
@@ -54,6 +58,26 @@ func extractCommand(cmd string) string {
 	return cmd
 }
 
+func readGaugeValue(g prometheus.Gauge) float64 {
+	var m *io_prometheus_client.Metric
+	err := g.Write(m)
+	if err != nil {
+		// How should I handle failure here?
+		return 0
+	}
+	return m.Counter.GetValue()
+}
+
+func (s MonitoredSync) UpdatePeakConnection(active prometheus.Gauge) {
+	peak := redisprom.PeakActiveConnectionsGauge.WithLabelValues(s.Name)
+	current := readGaugeValue(active)
```

Comment:
I don't think we should read the gauge value.

Let's use something like https://github.com/reddit/baseplate.go/pull/554 instead?

Reply:
sure I can move to that once it is merged